corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js

```diff
@@ -171,6 +171,175 @@ hqDefine("hqwebapp/js/knockout_bindings.ko", ['jquery', 'knockout', 'jquery-ui/u
+                    }
+
+                    var next = first_row;
+                    for(var i = start; i <= end; i++) {
+                        next.addClass('selected-for-sort');
+                        next = next.next();
+                    }
+                } else {
+                    $(this).addClass("selected-for-sort").addClass('last-clicked')
+                           .siblings().removeClass('selected-for-sort').removeClass('last-clicked');
+                }
+            });
+
+            $(element).on('click', '.send-to-top', function (e) {
+                $(this).parent().parent().addClass("moving").siblings().removeClass('moving');
+
+                // update UI
+                var row = $(this).parent().parent();
+                var current_index = row[0].attributes['data-order'].value;
+                row.parent().prepend(row);
+
+                // update KO
+                list().unshift(list().splice(current_index, 1)[0]);
+            });
+
+            $(element).on('click', '.send-to-bottom', function (e) {
+                $(this).parent().parent().addClass("moving").siblings().removeClass('moving');
+
+                var row = $(this).parent().parent();
+                var current_index = row[0].attributes['data-order'].value;
+
+                var lastSelectedRowIndex = null;
+                for (var i = 0; i < list().length; i++) {
+                    if (list()[i].selected()) {
+                        lastSelectedRowIndex = i;
+                    }
+                }
+
+                if (current_index < lastSelectedRowIndex) {
+                    // Update UI
+                    $('.isSelectedForExport').addClass('selected-for-export-above-last');
+                    row.removeClass('selected-for-export-above-last');
+                    $('.selected-for-export-above-last:nth-child('
+                        + ($('.selected-for-export-above-last').length + 1) + ')').after(row);
+                    $('.selected-for-export-above-last').removeClass('selected-for-export-above-last');
+
+                    // Update KO
+                    var current_list_item = list().splice(current_index, 1)[0];
+                    list().splice(lastSelectedRowIndex, 0, current_list_item);
+                }
+            });
+
+            $(element).sortable({
+                delay: 150,
+                helper: function (e, item) {
+                    if (!item.hasClass('selected-for-sort')) {
+                        item.addClass('selected-for-sort').siblings().removeClass('selected-for-sort');
+                    }
+
+                    var elements = item.siblings('.selected-for-sort').detach();
+                    $('body').append(elements);
+
+                    return item;
+                },
+                stop: function (e, ui) {
+                    ui.item.after($('.selected-for-sort'));
+
+                    // TODO - get working
```

Comment:
I'd like to maintain the original order of selected rows, but currently the row that is clicked and dragged ends up first.